### PR TITLE
ROX-12076: Add continuous deployment for probe [9]

### PR DESCRIPTION
**Pull request by stehessel**
## Description

Add a deployment script to execute the Helm chart for the probe service. The stage deployment is done via CD in the stage deployment workflow. The pull secret is injected as a secret into the probe namespace and linked to a probe service account. The secret itself is stored in the AWS parameter store.

## Checklist (Definition of Done)

- [ ] ~Unit and integration tests added~
- [x] Added test description under `Test manual`
- [ ] ~Evaluated and added CHANGELOG.md entry if required~
- [ ] ~Documentation added if necessary (i.e. changes to dev setup, test execution, ...)~
- [x] CI and all relevant tests are passing
- [x] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] ~Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.~

## Test manual

Tested the pull secret injection on dev cluster.

### CODE REVIEW

File: .github/workflows/deploy-stage.yaml [hunk 1]
```diff
@@ -6,18 +6,24 @@ on:
...
 
 jobs:
-  deploy:
-    name: "Re-terraform stage clusters"
+  cancel:
+    name: Cancel previous runs
     runs-on: ubuntu-latest
     permissions:
       id-token: write
       contents: read
```
Comment (porridge) [inline .github/workflows/deploy-stage.yaml:14]: These permissions are needed to get a token for talking to AWS parameter store.
Now that there is a separate `terraform` step, I think these will either need to be moved there (and copied to the  `deploy-probe` step too), or moved up one level (if that makes sense w.r.t. the GH Actions workflow schema).

File: deploy/helm/probe/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    PROBE_IMAGE="quay.io/rhacs-eng/blackbox-monitoring-probe-service:2b0c84d"
+    ;;
+
+  *)
+    echo "Unknown environment ${ENVIRONMENT}"
+    exit 2
+    ;;
+esac
+
+function assert_environment() {
```
Comment (porridge) [inline deploy/helm/probe/deploy.sh:42]: Can I ask you to approve https://github.com/stackrox/acs-fleet-manager/pull/584/files and do the same here?

**Review by porridge — CHANGES_REQUESTED** [review]
I was going to ask whether we have instructions for updating the pull secret (since apparently the base64-encoded blob needs to go into parameter store?) but then I started wondering whether we could somehow get rid of the need for the pull secret altogether, since the code is public anyway, and other images we deploy on the data plane clusters tend to be public as well, right?
That would be one less thing to worry about.

Also a couple of comments inline.

File: .github/workflows/deploy-stage.yaml (hunk 1 above)
Comment (stehessel) [reply .github/workflows/deploy-stage.yaml:14]: Ah, somehow I thought these are only needed by the `cancel` step.

File: deploy/helm/probe/deploy.sh (hunk 2 above)
Comment (stehessel) [reply deploy/helm/probe/deploy.sh:42]: Sure

**Comment by stehessel** [comment]
The other images are not public. But we pull them from `quay.io/app-sre` instead of `quay.io/rhacs-eng`, for which there seems to be already a pull secret by default on the cluster. However, I think making the probe image public is a good idea. I'll ask someone with quay admin credentials if this can be done.

**Comment by kurlov** [comment]
BTW, we can push images under quay.io/app-sre org in app-interface (e.g. [public fleet manager image](https://console.redhat.com/openshift/token/show#)). There are already creds for that in JenkinsCI runners. We can introduce probe image build and push as a [dedicated step](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/data/services/acs-fleet-manager/cicd/jobs.yaml)

**Comment by porridge** [comment]
> The other images are not public. But we pull them from `quay.io/app-sre` instead of `quay.io/rhacs-eng`, for which there seems to be already a pull secret by default on the cluster.

Hm, I thought these were public too, at least https://quay.io/repository/app-sre/acs-fleet-manager?tab=tags opens just fine in a fresh incognito window. :thinking:

**Review by porridge — APPROVED** [review]

**Pull request merged**